framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -366,6 +368,35 @@ public AnnotatedTypeFactory(BaseTypeChecker checker) {
         }
     }
 
+    /**
+     * Issue an error and abort if any of the support qualifiers have @Target meta-annotations
+     * that contain something besides TYPE_USE or TYPE_PARAMETER. (@Target({}) is allowed)
+     */
+    private void checkSupportedQuals() {
+        for (Class<? extends Annotation> annotationClass : supportedQuals) {
+            ElementType[] elements = annotationClass.getAnnotation(Target.class).value();
+            String otherElementTypes = null;
+            for (ElementType element : elements) {
+                if (!(element.equals(ElementType.TYPE_USE)
+                        || element.equals(ElementType.TYPE_PARAMETER))) {
+                    // if there's an ElementType with an enumerated value of something other
+                    // than TYPE_USE or TYPE_PARAMETER then it isn't a valid qualifier
+                    if (otherElementTypes == null) {
+                        otherElementTypes = element.toString();
+                    } else {
+                        otherElementTypes += ", " + element.toString();
+                    }
+                }
+            }
+            if (otherElementTypes != null) {
+                StringBuffer buf = new StringBuffer("The @Target meta-annotation on ");
```

Comment:
I suggest to add "type annotation" at the end of this string (before the annotation class name is added)